src/main/java/com/shapesecurity/shift/es2018/parser/GenericParser.java

```diff
@@ -817,10 +842,19 @@ protected AssignmentTarget transformDestructuring(Expression node) throws JsErro
         if (node instanceof ObjectExpression) {
             ObjectExpression objectExpression = (ObjectExpression) node;
             ArrayList<AssignmentTargetProperty> properties = new ArrayList<>();
-            for (ObjectProperty p : objectExpression.properties) {
-                properties.add(this.transformDestructuring(p));
+            Maybe<AssignmentTarget> rest = Maybe.empty();
+            ImmutableList<ObjectProperty> objectProperties = objectExpression.properties;
+            if (objectProperties.length > 0) {
+                ObjectProperty last = ((NonEmptyImmutableList<ObjectProperty>) objectProperties).last();
+                if (last instanceof SpreadProperty) {
+                    rest = Maybe.of(this.transformDestructuring(((SpreadProperty) last).expression));
+                    objectProperties = ((NonEmptyImmutableList<ObjectProperty>) objectProperties).init();
+                }
+                for (ObjectProperty p : objectProperties) {
```

Comment:
What if these contain other `SpreadProperty`s? Is it going to throw? What kind of failure message would that produce? Do we have a test that covers that?

Reply:
It will throw "invalid left-hand side in assignment", and is covered by [this test](https://github.com/shapesecurity/shift-java/blob/8f85220ce82892cc4f4d907b08d9975cf9dcb5a3/src/test/java/com/shapesecurity/shift/es2018/parser/destructuring/assignment/ObjectAssignmentTargetTest.java#L120).

(It's the same error you'd get for, for example, `({x: 0} = 0})` or `0 = 1`.)